addons/web/static/src/js/core.js

```diff
@@ -757,6 +757,23 @@ instance.web.unblockUI = function() {
     return $.unblockUI.apply($, arguments);
 };
 
+var currencies = {};
+
+//Load currencies information needed for monetary widget
+instance.web.load_currency = function() {
+    currencies = {};
+    return (new openerp.web.Model("res.currency").query(["symbol", "position", "decimal_places"]).filter().all())
+            .then(function(value) {
+                _.each(value, function(k,v){
+                    currencies[k.id] = {'symbol': k.symbol, 'position': k.position, 'digits': [69,k.decimal_places]};
+                });
+            });
+};
+
+instance.web.get_currency = function(currency_id) {
+    return currencies[currency_id];
+};
```

Comment:
Why not make all the currencies stuff part of the session, same as translations?